annotation-processor/src/main/java/org/hibernate/validator/ap/checks/ConstraintCheckFactory.java

```diff
@@ -189,7 +201,9 @@ public ConstraintCheckFactory(Types typeUtils, ConstraintHelper constraintHelper
 						new TargetCheck( annotationApiHelper ),
 						new ConstraintValidatorCheck( constraintHelper, annotationApiHelper ),
 						new AnnotationTypeMemberCheck( annotationApiHelper, typeUtils ),
-						new CrossParameterConstraintCheck( annotationApiHelper, constraintHelper, typeUtils )
+						new CrossParameterConstraintCheck( annotationApiHelper, constraintHelper, typeUtils ),
+						new AnnotationUserMessageCheck( annotationApiHelper ),
+						new AnnotationDefaultMessageCheck( annotationApiHelper, elementUtils )
```

Comment:
Ok, that looks better now. Only I am wondering why `AnnotationUserMessageCheck` is given here? My thinking is that there should be one check for the default message and another one for messages given when applying a constraint. I don't think the latter should also be applied for the default value, or should it?